theano/gpuarray/type.py

```diff
@@ -230,12 +239,10 @@ def filter(self, data, strict=False, allow_downcast=None):
                                         converted_data,
                                         force_same_dtype=False):
                     data = converted_data
-                    data = gpuarray.array(data, context=self.context)
 
             up_dtype = scalar.upcast(self.dtype, data.dtype)
             if up_dtype == self.dtype:
-                data = gpuarray.array(data, dtype=self.dtype, copy=False,
-                                      context=self.context)
+                data = numpy.array(data, dtype=self.dtype, copy=False)
```

Comment:
If data is on the GPU, then we will transfer it back to the CPU, to move it to the GPU again.